# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1400160781 -7200
# Node ID 645bd06b472cb8342473fd1574bc60cf546b9e10
# Parent  16d7f45b176b8d862a3923fcd93c93254133ce9d
x86/setup: resync the boot stack 8 bytes at a time

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>

diff -r 16d7f45b176b -r 645bd06b472c xen/arch/x86/setup.c
--- a/xen/arch/x86/setup.c
+++ b/xen/arch/x86/setup.c
@@ -906,7 +906,7 @@ void __init noreturn __start_xen(unsigne
 
             /* Re-sync the stack and then switch to relocated pagetables. */
             asm volatile (
-                "rep movsb        ; " /* re-sync the stack */
+                "rep movsq        ; " /* re-sync the stack */
                 "movq %%cr4,%%rsi ; "
                 "andb $0x7f,%%sil ; "
                 "movq %%rsi,%%cr4 ; " /* CR4.PGE == 0 */
@@ -914,7 +914,7 @@ void __init noreturn __start_xen(unsigne
                 "orb $0x80,%%sil  ; "
                 "movq %%rsi,%%cr4   " /* CR4.PGE == 1 */
                 : : "r" (__pa(idle_pg_table)), "S" (cpu0_stack),
-                "D" (__va(__pa(cpu0_stack))), "c" (STACK_SIZE) : "memory" );
+                "D" (__va(__pa(cpu0_stack))), "c" (STACK_SIZE / 8) : "memory" );
 
             bootstrap_map(NULL);
         }
